python/cuml/dask/ensemble/base.py

```diff
@@ -168,6 +171,26 @@ def _concat_treelite_models(self):
             TreeliteModel.free_treelite_model(tl_handle)
         return model
 
+    def _partial_inference(self, X, delayed, **kwargs):
+        data = DistributedDataHandler.create(X, client=self.client)
+        self.datatype = data.datatype

```

Comment:
Wouldn't like to set the datatype during inference. We'd expect inference not to modify the self object.

Reply:
Sure, I'll also remove it from `_predict_using_fil`.